### PR TITLE
Notebook Content Refining

### PR DESCRIPTION
**Pull request by gautamsarawagi**
@inodb @alisman 

I have added updated the notebook content to gives users a data preview in a suitable manner as follows: 

image

The sample code was as follows: 

# Install necessary library
%pip install -q seaborn

```
# Import libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Check if the file is empty
try:
    # Load the data from the CSV file
    df = pd.read_csv("metastatic_solid_tumors_mich_2017.csv")

    # Proceed only if the DataFrame is not empty
    if not df.empty:
        # Create a pivot table for the oncoprint
        pivot = df.pivot_table(index="hugoGeneSymbol", columns="sampleId", values="alterationType", aggfunc="first")

        # Check if the pivot table has data
        if not pivot.empty:
            # Set up the figure for plotting
            plt.figure(figsize=(20, 10))

            # Plot the oncoprint heatmap
            sns.heatmap(pivot.notnull(), cmap="Blues", cbar=False, linewidths=0.5, linecolor="black")

            # Set plot title and labels
            plt.title("Oncoprint")
            plt.xlabel("Samples")
            plt.ylabel("Genes")

            # Adjust tick labels
            plt.xticks(rotation=90)
            plt.yticks(rotation=0)

            # Adjust layout for better fit
            plt.tight_layout()

            # Display the plot
            plt.show()
        else:
            print("No data available for plotting.")
    else:
        print("The CSV file is empty.")
except pd.errors.EmptyDataError:
    print("The CSV file is empty or has no columns to parse.")
```

**Also added a Instruction.md file in the root directory** to give the knowledge about the data: 

image

Would really like to discuss further.

### CODE REVIEW

File: lite/lab/notebookContent.js [hunk 1]
```diff
@@ -8,76 +8,63 @@ export const createNotebookContent = ({ filename }) => ({
...
-                'ax.legend()\n',
-                'plt.title("3D Scatter Plot of Normalized Mutation Data")\n',
+                'import seaborn as sns\n',
+                '\n',
+                
+                '# Check if the file is empty\n',
+                'try:\n',
+                '    # Load the data from the CSV file\n',
+                `    df = pd.read_csv("${filename}")\n`,
+                '\n',
```
Comment (inodb) [inline lite/lab/notebookContent.js]: For the Python code, I prefer early return style on errors: https://softwareengineering.stackexchange.com/questions/18454. Avoids getting a lot of indenting

File: lite/files/Instructions.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+
+# Instructions for Using Oncoprint Data
+## Overview
+This notebook provides a visualization of mutation data using an oncoprint heatmap. This visualization helps in understanding the presence and distribution of genetic alterations across various samples.
```
Comment (inodb) [inline lite/files/Instructions.md]:
```suggestion
This notebook provides a visualization of mutation data using a heatmap. This visualization shows the presence of genetic alterations in each gene across the samples.
```

File: lite/api/contents/all.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (inodb) [inline lite/api/contents/all.json:1]: Should this file be in the repo? Same question for `.jupyterlite.doit.db`. They seem like temp files?

**Review by inodb — COMMENTED** [review]
@gautamsarawagi Awesome! Great improvement!

I noticed one other small thing in the title `CbioPortal` vs `cBioPortal`. Is that an easy fix?

File: lite/files/Instructions.md (hunk 2 above)
Comment (inodb) [reply lite/files/Instructions.md]: bit subtle but the OncoPrint heatmap has a lot more visual glyphs etc, calling it a "vanilla" heatmap makes it a bit more clear users' shouldn't expect the full OncoPrint functionality

File: lite/api/contents/all.json (hunk 3 above)
Comment (gautamsarawagi) [reply lite/api/contents/all.json:1]: @inodb

1. The `all.json` file in the content folders actually generates automatically during the build process when we run:
`jupyter lite build`.
Actually behind the scenes:
  
    _The build process indexes the contents found in {lite-dir}/files/ or any content roots specified via the CLI flag --contents or in jupyter_lite_config.json under #/LiteBuildConfig/contents. The indexed content is then saved in the output directory under {output-dir}/api/contents/{subdir?}/all.json._
    
    Refrence: https://jupyterlite.readthedocs.io/en/latest/howto/content/files.html#content-with-the-cli

2. **But we can definately delete the** `.jupyterlite.doit.db`. It is a temp file.

File: lite/lab/notebookContent.js (hunk 1 above)
Comment (gautamsarawagi) [reply lite/lab/notebookContent.js]: Sure let me just update the code with better indentation.

Comment (gautamsarawagi) [reply lite/lab/notebookContent.js]: And will also fix the typo in the tile of the project.

File: lite/api/contents/all.json (hunk 3 above)
Comment (gautamsarawagi) [reply lite/api/contents/all.json:1]: @inodb I suppose we cannot delete the `.jupyterlite.doit.db` also.

ref: https://discourse.jupyter.org/t/jupyter-lite-json-not-taken-into-account/21717.

I experinced the same issue.

There is also an open issue in jupyterlite regarding the same: https://github.com/jupyterlite/jupyterlite/issues/1194

Comment (inodb) [reply lite/api/contents/all.json:1]: ok thanks for checking and explaining @gautamsarawagi !

File: lite/lab/notebookContent.js (hunk 1 above)
Comment (inodb) [reply lite/lab/notebookContent.js]: Thank you!

**Review by inodb — APPROVED** [review]
Looks great - thanks so much!

**Comment by gautamsarawagi** [comment]
> Looks great - thanks so much!

Thanks a lot @inodb. I am still loking for the way to remove the `.jupyterlite.doit.db` file. This should be a temp file. Will keep you updated on this.

**Pull request merged**